dev/bots/test.dart

```diff
@@ -482,7 +482,7 @@ Future<void> _runAndroidPreviewIntegrationToolTests() async {
   final List<String> allTests = Directory(path.join(_toolsPath, 'test', 'android_preview_integration.shard'))
       .listSync(recursive: true).whereType<File>()
       .map<String>((FileSystemEntity entry) => path.relative(entry.path, from: _toolsPath))
-      .toList();
+      .where((String testPath) => path.basename(testPath).endsWith('.dart')).toList();
```

Comment:
What about:

```suggestion
      .where((String testPath) => path.basename(testPath).endsWith('_test.dart')).toList();
```

Reply:
Would you prefer that filter? For context the change from `_runIntegrationToolTests` was intentional because this test had historically been excluded from running without anyone noticing, but I don't have a strong preference and can change if you recommend

Reply:
since the default behavior of `dart test` is to only load files ending in `_test.dart`, by convention we have various helper libraries under //flutter_tools/test that should NOT be run by `dart test`, such as https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/dap/mocks.dart.

I don't feel too strongly though, it's just if someone wanted to add such a helper in the future, they'd need to update this line.

Reply:
To put another way, if there's a test file with a non *not* ending in _test.dart, I would call that a bug that the test was named wrong, and it wouldn't run locally with an invocation of `dart test`, and we should fix that, rather than have the CI runner actually run it.

If you wanted to be able to verify that we don't have any test files that don't run on CI because of this, the only way I could think of to do that would be to add a check to `//dev/bots/analyze.dart` and have an allow-listed of the above-mentioned test helper libraries that are not themselves test entrypoints.

Reply:
changed to suggested, I think it is reasonable that people would expect it to work like the one it very clearly based off of regardless

Reply:
> To put another way, if there's a test file _not_ ending in _test.dart, I would call that a bug that the test was named wrong, and it wouldn't run locally with an invocation of `dart test`, and we should fix that, rather than have the CI runner actually run it.
> 
> If you wanted to be able to verify that we don't have any test files that don't run on CI because of this, the only way I could think of to do that would be to add a check to `//dev/bots/analyze.dart` and have an allow-listed of the above-mentioned test helper libraries that are not themselves test entrypoints.

This comment was really helpful for me to get the context. I was one of the ones that encouraged gray to change the implementation.